net/udprelay/server.go

```diff
@@ -100,7 +157,7 @@ type ServerEndpoint struct {
 	// can be considered one and the same. If ServerDisco is equal, but
 	// LamportID is unequal, LamportID comparison determines which
 	// ServerEndpoint was allocated most recently.
-	LamportID uint64
+	LamportID ServerEndpointLamportID
```

Comment:
Alternatively, you could do:
```go
LamportID uint64 `json:",string"`
```
which will always treat this a JSON string containing a JSON number.

The benefit is less boilerplate declaring the `ServerEndpointLamportID` type, but at the risk of other usages of this ID accidentally using the JSON number representation (which would be the default).